cmd/vic-machine/create/create.go

```diff
@@ -818,28 +814,35 @@ func (c *Create) processNetwork(network *data.NetworkConfig, netName, pgName, st
 		}
 	}(network)
 
-	network.Destinations, network.Gateway, err = parseGatewaySpec(gateway)
-	if err != nil {
-		return fmt.Errorf("Invalid %s network gateway: %s", netName, err)
-	}
-	ipAddr, ipNet, err := net.ParseCIDR(staticIP)
-	if err != nil {
-		return fmt.Errorf("Provided %s network IP address %s has a wrong format", netName, staticIP)
+	if gateway != "" {
+		network.Destinations, network.Gateway, err = parseGatewaySpec(gateway)
+		if err != nil {
+			return fmt.Errorf("Invalid %s network gateway: %s", netName, err)
+		}
 	}
 
-	network.IP.IP = ipAddr
-	network.IP.Mask = ipNet.Mask
+	if staticIP != "" {
```

Comment:
I think you can parse staticIP first, and then parse gateway if it's not empty, and then you can merge two `if gateway != ""` statements into one section.